### PR TITLE
Add tests of unmarshaling literals to watch.py

### PR DESCRIPTION
**Pull request by jiahuif**
see #171
see kubernetes-client/python#982
see kubernetes-client/python#983

### CODE REVIEW

**Comment by jiahuif** [comment]
/cc @roycaihw 
Failing tests is intended as we are adding tests to reveal a bug.

File: watch/watch_test.py [hunk 1]
```diff
@@ -250,6 +250,24 @@ def test_unmarshal_with_custom_object(self):
         self.assertEqual("1", event['object']['metadata']['resourceVersion'])
         self.assertEqual("1", w.resource_version)
 
+    # ref: gh-982
+    def test_unmarshal_with_object_literal(self):
```
Comment (roycaihw) [inline watch/watch_test.py]: can we have the tests submitted with the fix?

**Comment by roycaihw** [comment]
/close
in favor of https://github.com/kubernetes-client/python-base/pull/185

**Pull request closed without merging**